lib/private/BackgroundJob/Job.php

```diff
@@ -47,7 +48,7 @@ abstract class Job implements IJob {
 	protected $argument;
 
 	/**
-	 * @param JobList $jobList
+	 * @param IJobList $jobList
 	 * @param ILogger|null $logger
 	 */
 	public function execute($jobList, ILogger $logger = null) {
```

Comment:
```suggestion
	public function execute(IJobList $jobList, ILogger $logger = null) {
```

and then drop the whole PHPDoc because it does not add any value to the actual code?

Reply:
Declaration is not compatible with interface then.

Reply:
😭 and I guess then also updating the interface cause apps to break ... 😭

Reply:
Contravariance actually allows that. We can add `IJobList` typehint on the parent and all existing implementations or subclasses should be fine.

Ref https://www.php.net/manual/en/language.oop5.variance.php

Reply:
Tested and works. At least for parameters it works in 7.2+. For return types we need to wait for 7.4+. Let's then go for the type hints, because this enforces better code 🚀

Reply:
I added them to this PR.